src/pages/SubmissionDetails.test.js

```diff
@@ -0,0 +1,174 @@
+import '@testing-library/jest-dom'
+
+import { Ajax } from 'src/libs/ajax'
+import { SubmissionDetails } from 'src/pages/SubmissionDetails'
+import { act, fireEvent, render, screen, within } from '@testing-library/react'
+import { h } from 'react-hyperscript-helpers'
+
+jest.mock('src/libs/ajax')
+
+jest.mock('src/libs/notifications.js')
+
+jest.mock('src/libs/config', () => ({
+  ...jest.requireActual('src/libs/config'),
+  getConfig: jest.fn().mockReturnValue({})
+}))
+
+describe('Submission Details page', () => {
+
+  // SubmissionDetails component uses AutoSizer to determine the right size for table to be displayed. As a result we need to
+  // mock out the height and width so that when AutoSizer asks for the width and height of "browser" it can use the mocked
+  // values and render the component properly. Without this the tests will be break.
+  // (see https://github.com/bvaughn/react-virtualized/issues/493 and https://stackoverflow.com/a/62214834)
+  const originalOffsetHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetHeight')
+  const originalOffsetWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetWidth')
+
+  const runsData = {
+    runs: [
+      {
+        run_id: 'ea001565-1cd6-4e43-b446-932ac1918081',
+        state: 'Submitted',
+        submission_date: '2022-01-27T22:27:15.591Z',
+        workflow_url: 'https://abc.wdl',
+        last_modified_timestamp: '2022-01-28T22:30:05.291Z',
+        workflow_params: '[{"parameter_name":"workflow_input_foo","parameter_type":"String","source":{"type":"literal","entity_attribute":"helloworld"}},{"parameter_name":"workflow_input_foo_rating","parameter_type":"Int","source":{"type":"entity_lookup","entity_attribute":"entity_field_foo_rating"}}]'
+      },
+      {
+        run_id: 'b7234aae-6f43-405e-bb3a-71f924e09825',
+        state: 'Failed',
+        submission_date: '2022-07-14T22:22:15.591Z',
+        workflow_url: 'https://xyz.wdl',
+        last_modified_timestamp: '2022-07-14T23:14:25.791Z',
+        workflow_params: '[{"parameter_name":"workflow_input_foo","parameter_type":"String","source":{"type":"literal","entity_attribute":"helloworld"}},{"parameter_name":"workflow_input_foo_rating","parameter_type":"Int","source":{"type":"entity_lookup","entity_attribute":"entity_field_foo_rating"}}]'
+      }
+    ]
+  }
+
+  beforeAll(() => {
+    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', { configurable: true, value: 1000 })
+    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 800 })
+  })
+
+  beforeEach(() => {
+    const getRunsMethod = jest.fn(() => Promise.resolve(runsData))
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runs: {
+            get: getRunsMethod
```

Comment:
Should this now take in Run Set ID as an argument?

Reply:
I think this was the question I was trying to ask yesterday but didn't know how to frame it

Reply:
I think it should to reflect how we get the runs?

Reply:
Yes it should be same as the actual API being used in the components.